src/rust/engine/options/src/args.rs

```diff
@@ -143,25 +145,83 @@ impl Args {
     }
 }
 
+pub(crate) struct ArgsTracker {
+    unconsumed_flags: Mutex<HashMap<Scope, HashSet<String>>>,
+}
+
+impl ArgsTracker {
+    fn new(args: &Args) -> Self {
+        let mut unconsumed_flags = HashMap::new();
+        for arg in &args.args {
```

Comment:
We could have used `&'a str` references with the lifetime of the `Args` instance, but that seems like overkill to avoid a tiny bit of string copying. Command lines are usually very short, and even unusually long ones would only spend microseconds on these copies.